lib/markdownlint.js

```diff
@@ -1293,6 +1305,12 @@ module.exports = markdownlint;
  * @property {RuleConfiguration} config Rule configuration.
  */
 
+/**
+ * Rule configuration object.
+ *
+ * @typedef {boolean | Object} RuleConfiguration Rule configuration.
+ */
+
```

Comment:
Since we are no longer using this type as part of the `Configuration`, I moved it next to the only type definition that uses it.